src/Tasks/Microsoft.NET.Build.Tasks/LockFileExtensions.cs

```diff
@@ -165,10 +141,14 @@ static int IndexOfWhiteSpace(string s)
 
             foreach (var group in lockFile.ProjectFileDependencyGroups)
             {
-                foreach (string dependency in group.Dependencies)
+                if (string.IsNullOrEmpty(group.FrameworkName) || string.IsNullOrEmpty(frameworkAlias) || group.FrameworkName.Equals(frameworkAlias) || 
+                    NuGetUtils.ParseFrameworkName(group.FrameworkName.Split('/').First()).DotNetFrameworkName.Equals(NuGetUtils.ParseFrameworkName(frameworkAlias).DotNetFrameworkName))
```

Comment:
Do we need to check if `group.FrameworkName` or `frameworkAlias` are empty?  Shouldn't they always have values?

Also, I think that eventually we should not need to parse these to compare them.  I think that eventually the dependency group's FrameworkName should match the target framework alias exactly.  However, currently it doesn't.  For now I would factor out the match calculation into a helper method to make it a bit easier to change later.

@nkolev92 @zkat Do you agree that when NuGet supports generalized target framework aliasing, that the key for the dependency group would change to be the target framework alias?  Is there an issue for that we should link?

Reply:
The target framework isn't marked as a required input for one of the tasks that uses this extension, so this guards against the case where it isn't provided. Without it we're breaking several existing tests.

Reply:
> Do you agree that when NuGet supports generalized target framework aliasing, that the key for the dependency group would change to be the target framework alias? Is there an issue for that we should link?

Not sure what the key is in this case. Can you please elaborate a bit?
Probably related https://github.com/NuGet/Home/issues/5154